File: src/Male.java
import java.util.Objects;

public class Male extends Family {
    private String nameWife;

    public Male(String firstName, String secondName, String nameFather, String nameMother, int age, String nameWife) {
        super(firstName, secondName, nameFather, nameMother, age);
        this.nameWife = nameWife;
    }

    public Male(String firstName, String secondName, String nameFather, String nameMother, int age) {
        super(firstName, secondName, nameFather, nameMother, age);
    }

    public Male() {

    }

    public String getNameWife() {
        return nameWife;
    }

    public void setNameWife(String nameWife) {
        this.nameWife = nameWife;
    }


    @Override
    public String toString() {
        return "Male{" +
                "nameWife='" + nameWife + '\'' +
                "} " + super.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Male male = (Male) o;
        return nameWife.equals(male.nameWife);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), nameWife);
    }
}

File: PythonOOP/FamilyPack/Male.py
from FamilyPack.MemberFamily import MemberFamily


class Male(MemberFamily):

    def __init__(self, first_name, second_name, name_father, name_mother, age, name_wife):
        super().__init__(first_name, second_name, name_father, name_mother, age)
        self._name_wife = name_wife

    @property
    def name_wife(self):
        return self._name_wife

    @name_wife.setter
    def name_wife(self, value):
        self._name_wife = value

    def __str__(self):
        return super().__str__() + '\n' + str(self.name_wife)
            